src/v/utils/delta_for.h

```diff
@@ -218,6 +232,32 @@ class deltafor_encoder {
         _cnt++;
     }
 
+    // State of the transaction
+    // The state can be used to append multiple rows
+    // and then commit or rollback.
+    struct tx_state {
+        using self_t = deltafor_encoder<TVal, DeltaStep>;
+        self_t uncommitted;
+
+        void add(const row_t& row) { uncommitted.add(row); }
+    };
+
+    // Create tx-state object and start transaction
+    //
+    // Only one transaction at a time is supported but this
+    // is not enforced. Abandoning tx_state object is ok (this
```

Comment:
It's designed to be lightweight. Before we enable fail on bad_alloc every update to the manifest has to be transactional. And because of the way it's used we only have one tx at a time.